shared/runtime/pyexec.c

```diff
@@ -113,7 +114,11 @@ STATIC int parse_compile_execute(const void *source, mp_parse_input_kind_t input
         }
 
         // execute code
-        mp_hal_set_interrupt_char(CHAR_CTRL_C); // allow ctrl-C to interrupt us
+        if (exec_flags & EXEC_FLAG_NO_INTERRUPT) {
+            mp_hal_set_interrupt_char(-1);
```

Comment:
No need to call this here, it should be `-1` on entry to this function (and if it's not, that's a bug and needs to be fixed elsewhere).  So this can just be:
```c
if (!(exec_flags & EXEC_FLAG_NO_INTERRUPT)) {
    mp_hal_set_interrupt_char(CHAR_CTRL_C);
}
```